habitat-lab/habitat/sims/habitat_simulator/object_state_machine.py

```diff
@@ -0,0 +1,468 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, List, Union
+
+import magnum as mn
+
+import habitat.sims.habitat_simulator.sim_utilities as sutils
+import habitat_sim
+from habitat.sims.habitat_simulator.debug_visualizer import (
+    draw_object_highlight,
+)
+from habitat.sims.habitat_simulator.metadata_interface import MetadataInterface
+from habitat_sim.physics import ManagedArticulatedObject, ManagedRigidObject
+
+##################################################
+# Supporting utilities for getting and setting metadata values in ManagedObject "user_defined" Configurations.
+##################################################
+
+
+def get_state_of_obj(
+    obj: Union[ManagedArticulatedObject, ManagedRigidObject],
+    state_name: str,
+) -> Any:
+    """
+    Try to get the specified state from an object's "object_states" user_defined metadata.
+
+    :param obj: The ManagedObject.
+    :param state_name: The name/key of the object state property to query.
+
+    :return: The state value (variable type) or None if not found.
+    """
+
+    if "object_states" in obj.user_attributes.get_subconfig_keys():
+        obj_states_config = obj.user_attributes.get_subconfig("object_states")
+        if obj_states_config.has_value(state_name):
+            return obj_states_config.get(state_name)
+    return None
+
+
+def set_state_of_obj(
+    obj: Union[ManagedArticulatedObject, ManagedRigidObject],
+    state_name: str,
+    state_val: Any,
+) -> None:
+    """
+    Set the specified state in an object's "object_states" user_defined metadata.
+
+    :param obj: The ManagedObject.
+    :param state_name: The name/key of the object state property to set.
+    :param state_val: The value of the object state property to set.
+    """
+
+    user_attr = obj.user_attributes
+    obj_state_config = None
+    if "object_states" not in user_attr.get_subconfig_keys():
+        obj_state_config = (
+            habitat_sim._ext.habitat_sim_bindings.Configuration()
+        )
+    else:
+        obj_state_config = user_attr.get_subconfig("object_states")
```

Comment:
You only need this line.  It will either retrieve or create and then retrieve a config with the given name.